### DISCUSSION TITLE
MCMC in parallel with different datasets

### DISCUSSION THREAD

**Discussion by jeremiecoullon** [blackjax-devs/blackjax#139] [Q&A] [answered]
Hello!

However I'm wondering if it's possible to run several MCMC in parallel, with each targettng the same posterior with a different dataset. Is this something that is possible with BlackJax, or does the code running on each device have to be the same?

I'm asking as the kernel needs to be built before defining the sampler. So doing a vmap/pmap over different datasets might be possible, but the function being mapped would need to build the kernel..

Thanks!

**Comment by junpenglao** [comment, marked as answer]
Yes this should be possible with vmapping the inference/sampling routine, something like:

```python
# define a function that runs inference on 1 dataset:
@jax.vmap
def run_inference(dataset):
   ...
```

the kernel building will encapsulate within the `run_inference` function

**Reply by jeremiecoullon** [reply]
Nice! I assumed that the kernel building was not jittable. I'll try this out :)

Thanks very much!
